lib/APPSensorFusion/App.cpp

```diff
@@ -159,6 +168,22 @@ void App::sendSensorData() const
     (void)m_smpServer.sendData(m_smpChannelIdSensorData, reinterpret_cast<uint8_t*>(&payload), sizeof(payload));
 }
 
+void App::sendEndLineDetectionSignal()
+{
+    DrivingState::LineStatus lineStatus = DrivingState::getInstance().getLineStatus();
+
+    /* Send only if a new End Line has been detected. */
+    if ((DrivingState::LINE_STATUS_FIND_END_LINE == m_lastLineDetectionStatus) &&
+        (DrivingState::LINE_STATUS_END_LINE_DETECTED == lineStatus))
+    {
+        EndLineFlag payload;
+        payload.isEndLineDetected = true;
+        (void)m_smpServer.sendData(m_smpChannelIdEndLine, reinterpret_cast<uint8_t*>(&payload), sizeof(payload));
```

Comment:
Why is the return value not checked? What happens if one line is "skipped" because of communication?

Reply:
That case will be handled in the Monitoring Script.